packages/pipeline/src/scripts/pull_paradex_orderbook_snapshots.ts

```diff
@@ -0,0 +1,93 @@
+// tslint:disable:no-console
+import { BigNumber } from '@0x/utils';
+import * as R from 'ramda';
+import { Connection, ConnectionOptions, createConnection } from 'typeorm';
+
+import {
+    PARADEX_SOURCE,
+    ParadexActiveMarketsResponse,
+    ParadexMarket,
+    ParadexSource,
+    ParadexTokenInfo,
+    ParadexTokenInfoResponse,
+} from '../data_sources/paradex';
+import { TokenOrderbookSnapshot as TokenOrder } from '../entities';
+import * as ormConfig from '../ormconfig';
+import { parseParadexOrders } from '../parsers/paradex_orders';
+import { handleError } from '../utils';
+
+// Number of orders to save at once.
+const BATCH_SAVE_SIZE = 1000;
+
+let connection: Connection;
+
+(async () => {
+    connection = await createConnection(ormConfig as ConnectionOptions);
+    const apiKey = process.env.PARADEX_API_KEY;
+    if (apiKey === undefined) {
+        throw new Error('Missing required env var: PARADEX_API_KEY');
+    }
+    const paradexSource = new ParadexSource(apiKey);
+    const tokenOrderRepository = connection.getRepository(TokenOrder);
+    const markets = await paradexSource.getActiveMarketsAsync();
+    const tokenInfoResponse = await paradexSource.getTokenInfoAsync();
+    const extendedMarkets = addTokenAddresses(markets, tokenInfoResponse);
+    await Promise.all(
+        extendedMarkets.map(async (market: ParadexMarket) => getAndSaveMarketOrderbook(paradexSource, market)),
+    );
+    process.exit(0);
+})().catch(handleError);
+
+/**
+ * Extend the default ParadexMarket objects with token addresses.
+ * @param markets An array of ParadexMarket objects.
+ * @param tokenInfoResponse An array of ParadexTokenInfo containing the addresses.
+ */
+function addTokenAddresses(
+    markets: ParadexActiveMarketsResponse,
+    tokenInfoResponse: ParadexTokenInfoResponse,
+): ParadexMarket[] {
+    const extractSymbolAddress = (acc: Map<string, string>, tokenInfo: ParadexTokenInfo): Map<string, string> => {
+        acc.set(tokenInfo.symbol, tokenInfo.address);
```

Comment:
`acc.set` mutates the map. If we are going to use mutation here, we don't need to use `reduce`. We can just do something like:

```ts
const symbolToAddress = new Map<string, string>();
tokenInfoResponse.forEach(tokenInfo => symbolToAddress.set(tokenInfo.symbol, tokenInfo.address));
```